tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,1923 @@
+        private const string KeyVaultName = "keyvault-dotnetprovider";
+        private const string ResourceGroupName = "dotnetprovider-integrationtest";
+
+        /// <summary>
+        /// Class to hold test-specific key information
+        /// </summary>
+        private class TestContext
+        {
+            public string KeyPrefix { get; set; }
+            public string SentinelKey { get; set; }
+            public string FeatureFlagKey { get; set; }
+            public string KeyVaultReferenceKey { get; set; }
+            public string SecretName { get; set; }
+            public string SecretValue { get; set; }
+        }
+
+        // Client for direct manipulation of the store
+        private ConfigurationClient _configClient;
+
+        // Client for Key Vault operations
+        private SecretClient _secretClient;
+
+        // Connection string for the store
+        private string _connectionString;
+
+        // Endpoints for the resources
+        private Uri _appConfigEndpoint;
+        private Uri _keyVaultEndpoint;
+
+        // Track resources created by tests for cleanup
+        private readonly HashSet<string> _createdConfigKeys = new HashSet<string>();
+        private readonly HashSet<string> _createdSecretNames = new HashSet<string>();
+        private readonly HashSet<string> _createdSnapshotNames = new HashSet<string>();
+
+        /// <summary>
+        /// Gets a DefaultAzureCredential for authentication.
+        /// </summary>
+        private DefaultAzureCredential GetCredential()
+        {
+            return new DefaultAzureCredential(new DefaultAzureCredentialOptions
+            {
+                ExcludeSharedTokenCacheCredential = true
+            });
+        }
+
+        /// <summary>
+        /// Gets the current subscription ID by reading from the JSON file created by the PowerShell script.
+        /// NOTE: The PowerShell script must be run manually before running the tests.
+        /// </summary>
+        private string GetCurrentSubscriptionId()
+        {
+            string subscriptionIdFromEnv = Environment.GetEnvironmentVariable("AZURE_SUBSCRIPTION_ID");
+
+            if (!string.IsNullOrEmpty(subscriptionIdFromEnv))
+            {
+                return subscriptionIdFromEnv;
+            }
+
+            // Read the JSON file created by the script
+            string jsonPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "Integration", SubscriptionJsonPath);
+
+            if (!File.Exists(jsonPath))
+            {
+                throw new InvalidOperationException($"Subscription JSON file not found at {jsonPath}. Run the GetAzureSubscription.ps1 script first.");
+            }
+
+            string jsonContent = File.ReadAllText(jsonPath);
+
+            using JsonDocument doc = JsonDocument.Parse(jsonContent);
+            JsonElement root = doc.RootElement;
+
+            bool success = root.GetProperty("Success").GetBoolean();
```

Comment:
I don't understand why this is valuable?

Reply:
yea it can be removed